google-cloud-logging/src/main/java/com/google/cloud/logging/MonitoredResourceUtil.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.logging;
+
+import com.google.cloud.MetadataConfig;
+import com.google.cloud.MonitoredResource;
+import com.google.common.base.Strings;
+import java.util.List;
+
+abstract class MonitoredResourceUtil {
+
+  private static final String defaultResourceType = "global";
+
+  private static void addResourceLabels(
+      String resourceType, MonitoredResource.Builder resourceBuilder) {
+    String[] labels;
+    switch (resourceType) {
+      case "gae_app_flex":
+        labels = new String[] {"module_id", "version_id", "zone", "instance_id"};
+        break;
+      case "gae_app_standard":
+        labels = new String[] {"module_id", "version_id"};
+        break;
+      case "container":
+        labels = new String[] {"cluster_name", "zone"};
+        break;
+      case "gce_instance":
+        labels = new String[] {"instance_id", "zone"};
+        break;
```

Comment:
I think I would prefer pre-populating a map of resource type -> string[], and then doing a get() in this function.